pkg/certificates/reconciler/certificates.go

```diff
@@ -138,22 +158,33 @@ func parseAndValidateSecret(secret *corev1.Secret, shouldContainCaCert bool) (*x
 		}
 	}
 
-	caCert, caPk, err := certificates.ParseCert(certBytes, pkBytes)
+	cert, caPk, err := certificates.ParseCert(certBytes, pkBytes)
 	if err != nil {
 		return nil, nil, err
 	}
-	if err := certificates.ValidateCert(caCert, rotationThreshold); err != nil {
+	if err := certificates.ValidateCert(cert, rotationThreshold); err != nil {
 		return nil, nil, err
 	}
-	return caCert, caPk, nil
+
+OUTER:
+	for _, san := range sans {
+		for _, certSan := range cert.DNSNames {
+			if strings.EqualFold(certSan, san) {
+				continue OUTER
+			}
+		}
+		return nil, nil, fmt.Errorf("missing san %q", san)
+	}
```

Comment:
This feels like it would be more efficient with a set-union pattern:

```golang
sanSet := sets.New[string]()
for _, san := range sans {
  sanSet.Insert(strings.ToLower(san))
}
certSet := sets.New[string]()
for _, certSan := range cert.DNSNames {
  sanSet.Insert(strings.ToLower(certSan)
}
if !sanSet.Equal(certSet) {
  return nil, nil, fmt.Errorf("Unexpected SANs")
}
```

This covers both _missing_ and _accidentally included_ SANs.  If we can guarantee that one or both sets of SANs are lowercase already, then you could simply write:

```golang
sanSet := sets.New(sans...)
certSet := sets.New(cert.DNSNames...)
```

Reply:
Since we create one of the lists internally, we can make sure it is lower case.